DESIGN.md

```diff
@@ -0,0 +1,108 @@
+
+The directorylist file itself is simple, it creates the necessary root directories in WiX, then inside the last directory it calls out to Vanagon::Platform::Windows.generate_service_bin_dirs. This function does a two things for each service file:
+
+1. Generates the necessary directories underneath where the file is, these directory elements will have names and resolve to a path
+2. Generates one last element underneath the top direcotory with no name and bindir_id as the id element. This provides the necessary connection between bindir_id and the actual directory containing binaries. As discussed before because this element has no name it will resolve to it’s parent directory element, the top level directory.
+
+In cases where there is more than one bindir_id corresponding to the same place there is no problem, there will just be two nameless directory elements under that directory. This is allowed in WiX because they have no names.
+
+### INSTALLDIR
+Inside the directorylist file the last root directory defined has the unique id INSTALLDIR. Vanagon uses this name on purpose. Used correctly, INSTALLDIR provides the user the means to change the location of the final install of the MSI. Using WiX properties and the INSTALLDIR variable the user can override all root directories underneath INSTALLDIR with what the end user wants.
+
+Note that this is in no way the correct way for the vanagon user to override directorylist root install defaults. If the user wishes the default installation location to be something other than DRIVE:/ProgramFilesFolder/CompanyName/ProductName they should override the entire directorylist file with a file of the same name in their own project. Vanagon will default to using the one in their project.
+
+INSTALLDIR itself is not a WiX property or requirement, the name was a convention that vanagon uses.
+
+### Candle.exe
+Once we unpacked everything into the staging directory, run heat.exe on its contents and executed the vanagon erb functionality on all WiX files, we need to actually compile the WiX files. Candle.exe is akin to the actual compile step of compilers.
+
+The invocation of candle is very straightforward, with one caveat: as discussed earlier, we need to pass the preprocessor variable “AppSourcePath” for heat to use.
+
+More information on the Candle compiler:
+
+* [docs](http://wixtoolset.org/documentation/manual/v3/)
+* [tools](http://wixtoolset.org/documentation/manual/v3/overview/alltools.html)
+
+### Light.exe
+Light.exe is akin to the linking step in compilers. In the invocation of light we don’t use anything out of the ordinary, we simply pass it what the location the files exist in, and where the output file should go. The output of Light.exe is an actual MSI, so after this step vanagon has completed compilation.
```

Comment:
Mention that localisation is also fed in at this stage from localised string tables.